services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -338,7 +342,28 @@ public Response processLogin(@QueryParam("code") String code,
                     .setUriInfo(uriInfo);
             return protocol.cancelLogin(clientSession);
         }
-
+        
+        UserModel user = KeycloakModelUtils.findUserByNameOrEmail(session, realm, username);
```

Comment:
To make it consistent with other checks during login please move this to nextActionAfterAuthentication method and add a static method like what's done for isTotpConfigurationRequired on L378 of LoginActionsService